test/channels.jl

```diff
@@ -626,3 +626,20 @@ end
         @test n_avail(c) == 0
     end
 end
+
+# Issue #49507: stackoverflow in type inference caused by close(::Channel, ::Exception)
+@testset "close(::Channel, ::StackOverflowError)" begin
```

Comment:
I forgot I meant to say this should go into the `stack_overflow.jl` file, since the handling of this error is quite bad and sometimes ends up breaking the process (as I found out today trying to debug something completely unrelated). Do you mind making a PR to move it?

Reply:
Oh, interesting. Yeah, i'm happy to move it! 👍 PR incoming

Reply:
PR here: https://github.com/JuliaLang/julia/pull/49702